src/ng/compile.js

```diff
@@ -779,6 +779,14 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     return bindings;
   }
 
+  function assertFirstLetterIsLowerCased(name) {
```

Comment:
rename to `assertValidDirectiveName`. The function should also check that the directive name is not the empty string